app/controllers/api/pages_controller.rb

```diff
@@ -51,4 +94,11 @@ def shallow_errors(errors)
   def get_page
     @page = Page.find(params[:id])
   end
+
+  def set_language
+    @language ||= Language.find_by(code: params[:language])
```

Comment:
What about using the `!` version of `find_by`, since it returns 404 for you if record is not found. I don't think you need to provide a custom error message. wdt?